skorch/tests/test_net.py

```diff
@@ -1409,6 +1419,70 @@ def test_batch_size_neg_1_uses_whole_dataset(
         assert train_kwargs['batch_size'] == expected_train_batch_size
         assert valid_kwargs['batch_size'] == expected_valid_batch_size
 
+    def test_fit_lbfgs_optimizer(self, net, data):
+        X, y = data
+        net.set_params(optimizer=torch.optim.LBFGS)
+        net.set_params(batch_size=len(X))
+        net.fit(X, y)
+
+    def test_accumulator_that_returns_last_value(
+            self, net_cls, module_cls, data):
+        # We define a loss that is called 3 times and an accumulator
+        # that returns the last of those calls. We then test that the
+        # correct values were stored.
+        from skorch.utils import FirstStepAccumulator
+        side_effect = []
+
+        class Criterion3Calls:
+            """Call NLLLoss 3 times."""
+            loss = torch.nn.NLLLoss()
+            def __call__(self, *args, **kwargs):
+                loss = self.loss(*args, **kwargs)
+                side_effect.append(float(loss))
+                loss = self.loss(*args, **kwargs)
+                side_effect.append(float(loss))
+                loss = self.loss(*args, **kwargs)
+                side_effect.append(float(loss))
+                return loss
```

Comment:
This should be a custom optimizer instead of a criterion as the additional loss values do not contribute to the test.